src/AgOledDisplay.cpp

```diff
@@ -380,7 +380,33 @@ void OledDisplay::showNewFirmwareFailed(void) {
   do {
     DISP()->setFont(u8g2_font_t0_16_tf);
     setCentralText(20, "Firmware Update");
-    setCentralText(40, "Failed");
-    setCentralText(60, String("Retry after 24h"));
+    setCentralText(40, "fail, will retry");
+    // setCentralText(60, "will retry");
+  } while (DISP()->nextPage());
+}
+
+void OledDisplay::showNewFirmwareSkipped(void) {
+  if (isDisplayOff) {
+    return;
+  }
+
+  DISP()->firstPage();
+  do {
+    DISP()->setFont(u8g2_font_t0_16_tf);
+    setCentralText(20, "Firmware Update");
```

Comment:
Propose to rephrase to
Firmware update skipped

Reply:
If it's fit for a line we should reduce the font size.